gymnasium/dev_wrappers/time_aware_observation.py

```diff
@@ -0,0 +1,135 @@
+                otherwise return time as remaining timesteps before truncation
+        """
+        super().__init__(env)
+        self.flatten = flatten
+        self.normalize_time = normalize_time
+        self.max_timesteps = self._get_max_timesteps(env)
+        self.num_envs = getattr(env, "num_envs", 1)
+
+        if self.normalize_time:
+            self._get_time_observation = lambda: self.timesteps / self.max_timesteps
+            time_space = Box(0, 1, (self.num_envs,))
+        else:
+            self._get_time_observation = lambda: self.max_timesteps - self.timesteps
+            time_space = Box(0, self.max_timesteps, (self.num_envs,))
+
+        self.time_aware_observation_space = Dict(
+            obs=env.observation_space, time=time_space
+        )
+
+        if self.flatten:
+            self.observation_space = spaces.flatten_space(
+                self.time_aware_observation_space
+            )
+            self._observation_postprocess = lambda observation: spaces.flatten(
+                self.time_aware_observation_space, observation
+            )
+        else:
+            self.observation_space = self.time_aware_observation_space
+            self._observation_postprocess = lambda observation: observation
+
+    def observation(self, observation: ObsType):
+        """Adds to the observation with the current time information.
+
+        Args:
+            observation: The observation to add the time step to
+
+        Returns:
+            The observation with the time information appended to
+        """
+        time_observation = self._get_time_observation()
+        observation = OrderedDict(obs=observation, time=time_observation)
+
+        return self._observation_postprocess(observation)
+
+    def step(self, action: ActType):
+        """Steps through the environment, incrementing the time step.
+
+        In vectorized environments, if any of the environments
+        reach a terminal state, set the respective time value to zero.
+
+        Args:
+            action: The action to take
+
+        Returns:
+            The environment's step using the action.
+        """
+        self.timesteps += 1
+        observation, reward, terminated, truncated, info = super().step(action)
+
+        self.timesteps = jp.where(terminated, 0, self.timesteps)
+
+        return observation, reward, terminated, truncated, info
+
+    def reset(self, **kwargs):
+        """Reset the environment setting the time to zero.
+
+        Args:
+            **kwargs: Kwargs to apply to env.reset()
+
+        Returns:
+            The reset environment
+        """
+        self.timesteps = jp.zeros(self.num_envs)
+        return super().reset(**kwargs)
+
+    @staticmethod
+    def _get_max_timesteps(env: gym.Env):
+        """Get the max episode steps before truncation occurs.
+
+        Args:
+            env: the provided environment
+        """
+        return (
```

Comment:
Remove, this should be able to be simplified to just consider `gym.Env` not `VectorEnv`

Reply:
https://github.com/Farama-Foundation/Gymnasium/pull/165/commits/75d7978551caa9416c706e8c30c72888943b63ad